File: src/CSharp/482.LicenseKeyFormatting.cs

using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.LicenseKeyFormatting
{
public class Solution {
    public string LicenseKeyFormatting(string S, int K) {
        var n = S.Length;
        var count = 0;
        var chars = new List<char>();
        for(int i = 0; i < n; i++)
        {
            var j = n - i - 1;
            if (S[j] == '-')
                continue;
            if (S[j] >= 'a' && S[j] <= 'z')
                chars.Add((char)(S[j] - 'a' + 'A'));
            else
                chars.Add(S[j]);
            count++;
            if (count % K == 0)
                chars.Add('-');
        }
        var size = chars.Count;
        if (size > 0 && chars.Last() == '-')
            chars.RemoveAt(size - 1);
        chars.Reverse();
        return new string(chars.ToArray());
    }
}

}

File: src/Java/401-500/482.LicenseKeyFormatting.java

import java.awt.List;
import java.util.LinkedList;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public String licenseKeyFormatting(String S, int K) {
        char[] Sarr = S.toCharArray();
        int n = Sarr.length;
        int count = 0;
        int size = 0;
        LinkedList<Character> chars = new LinkedList<>();
        for(int i = 0; i < n; i++)
        {
            int j = n - i - 1;
            if (Sarr[j] == '-')
                continue;
            if (Sarr[j] >= 'a' && Sarr[j] <= 'z')
                chars.add((char)(Sarr[j] - 'a' + 'A'));
            else
                chars.add(Sarr[j]);
            count++;
            size++;
            if (count % K == 0){
                chars.add('-');
                size++;
            }
        }
        
        if (size > 0 && chars.getLast() == '-')
            chars.removeLast();
        Character[] charsarr = new Character[size];
        Collections.reverse(chars);
        chars.toArray(charsarr);
        return chars.stream()
                     .map(String::valueOf)
                     .collect(Collectors.joining());
    }
}